sympy/core/add.py

```diff
@@ -944,6 +944,15 @@ def _eval_subs(self, old, new):
                     return self.func(-new, coeff_self, coeff_old,
                                *[s._subs(old, new) for s in ret_set])
 
+            # c*(a + b).subs(a + b, x) -> c*x (c being the multiplicative constant)
+            c = self.as_content_primitive()[0]
+            if not c is S.One:
+                add_term = self.extract_multiplicatively(c)
+                if add_term is not None:
+                    expr = add_term._eval_subs(old, new)
+                    if expr is not None:
+                        return c*expr
```

Comment:
This function already looks more complicated than it should be. I'm not sure about adding this extra complexity.

Reply:
Also does `extract_multiplicatively` not just return the same thing as `self.as_content_primitive()[1]` (which you are discarding above)?

Reply:
Wow thanks for pointing this out here, this was my first attempt to fixed something related to the `core` module , hence was somewhat inexperienced back then . I'll make the changes , yeah even something like the following should do 

```
--- a/sympy/core/add.py
+++ b/sympy/core/add.py
@@ -946,12 +946,10 @@ def _eval_subs(self, old, new):

             # c*(a + b).subs(a + b, x) -> c*x (c being the multiplicative constant)
             c = self.as_content_primitive()[0]
+            add_term = self.as_content_primitive()[1]
             if not c is S.One:
-                add_term = self.extract_multiplicatively(c)
-                if add_term is not None:
-                    expr = add_term._eval_subs(old, new)
-                    if expr is not None:
-                        return c*expr
+                expr = add_term._eval_subs(old, new)
+                return c*expr
```

Reply:
I hope the latest commit looks better .

Reply:
@oscarbenjamin does this look better to you ?